api/Silk.Api/Middleware/InternalServerErrorWrapper.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentValidation;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Silk.Api.ApiResponses;
+
+namespace Silk.Api
+{
+	public sealed class InternalServerErrorWrapper 
+	{
+		private readonly RequestDelegate _next;
+		private readonly ILogger<InternalServerErrorWrapper> _logger;
+
+		public InternalServerErrorWrapper(RequestDelegate next, ILogger<InternalServerErrorWrapper> logger)
+		{
+			_next = next;
+			_logger = logger ?? throw new ArgumentNullException(nameof(logger));
+		}
+
+		public async Task Invoke(HttpContext context)
+		{
+			try
+			{
+				await _next.Invoke(context);
+			}
+			catch (Exception ex)
+			{
+				await HandleException(context, ex);
+			}
+		}
+
+		private async Task HandleException(HttpContext context, Exception ex)
+		{
```

Comment:
Don't have to change it now, but def before public api release, will prob want to limit what info is sent back when an error occurs in production (not to expose sensitive error details)

Reply:
This only returns back FluentValidation erorrs/What ASP.NET would be sending back anyway, so I don't think it's *too* bad? I've yet to see anything sensitive come out of it as of yet